Main revision of Mantid Workbench release notes for v6.4.0

sf1919 opened the pull request:
**Description of work.**

Main revision of Mantid Workbench release notes for v6.4.0

- Expanded on notes where necessary
- Amalgamated release notes relating to matplotlib v3.5 changes into a single note
- Added release notes that had been included after the switch to manual release notes and moved those files into Used folders as appropriate
- Added links
- Added an image for the the new cut viewer tool in Sliceviewer
- Spelling and grammar check

**To test:**

- remove `build/docs/doctrees`
- build the documents again and check there are no warnings
- Check that the links work
- Check the information is still correct
- Grammar/Spelling check

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Main revision of Mantid Workbench release notes for v6.4.0

- Expanded on notes where necessary
- Amalgamated release notes relating to matplotlib v3.5 changes into a single note
- Added release notes that had been included after the switch to manual release notes and moved those files into Used folders as appropriate
- Added links
- Added an image for the the new cut viewer tool in Sliceviewer
- Spelling and grammar check

ConorMFinn commented on docs/source/release/v6.4.0/mantidworkbench.rst:
```diff
@@ -8,46 +8,40 @@ Mantid Workbench Changes
-- New general setting: `Provide auto-completion suggestions in the script editor` for making auto-completion optional in the script editor
-- An additional memory bar has been added to the memory widget to enable users to see how much memory Mantid is using.
+- Added the :ref:`Fit Script Generator <Fit_Script_Generator-ref>` interface to the 'General' category of the interface menu.
+- Added a new general setting: ``Provide auto-completion suggestions in the script editor`` for making auto-completion optional in the script editor.
```
I'm not sure this should be monospaced, as it isn't referencing code?

sf1919 replied:
Whilst it is note code in the strictest of senses it is the new option listed on the setting tab for users to use. It makes it more obvious that this is what the user is looking for in settings. I've realised I missed the formatting for General so I'll add that in.

ConorMFinn replied:
True, I just think it looks slightly strange for such a long string. Maybe _"Provide auto-completion suggestions in the script editor"_ would look better. Though I'm not overly fussed either way

ConorMFinn commented on docs/source/release/v6.4.0/mantidworkbench.rst:
```diff
@@ -8,46 +8,40 @@ Mantid Workbench Changes
-- Removed the load ILL tab from the interface in "Indirect" -> "Tools". Users should use LoadILLIndirect algorithm or the Load button instead.
-- RebinnedOutput workspaces are now supported in the Superplot
-- Copying a spectrum in a MatrixWorkspace is now supported for ragged workspaces.
+- Removed the Load ILL tab from the interface in ``Indirect -> Tools``. Users should use :ref:`LoadILLIndirect<algm-LoadILLIndirect>` algorithm or the ``Load`` button instead.
```
```suggestion
- Removed the Load ILL tab from the interface in ``Indirect -> Tools``. Users should use the :ref:`LoadILLIndirect<algm-LoadILLIndirect>` algorithm or the ``Load`` button instead.
```

ConorMFinn commented on docs/source/release/v6.4.0/mantidworkbench.rst:
```diff
@@ -8,46 +8,40 @@ Mantid Workbench Changes
-- RebinnedOutput workspaces are now supported in the Superplot
-- Copying a spectrum in a MatrixWorkspace is now supported for ragged workspaces.
+- Removed the Load ILL tab from the interface in ``Indirect -> Tools``. Users should use :ref:`LoadILLIndirect<algm-LoadILLIndirect>` algorithm or the ``Load`` button instead.
+- ``RebinnedOutput`` workspaces are now supported in the Superplot
```
```suggestion
- ``RebinnedOutput`` workspaces are now supported in the Superplot.
```

ConorMFinn commented on docs/source/release/v6.4.0/mantidworkbench.rst:
```diff
@@ -61,14 +55,21 @@ Sliceviewer
+- Using the ``Log`` or ``SymmetricLog10`` colorbar normalisation options on masked data is now disabled to prevent an error.
+- Fixed a bug with peak selection causing a crash for 4D MD workspaces.
+- Peak addition now correctly assigns new peak ``HKL`` in nonorthogonal view.
+- Peak overlay is now disabled when non-Q axis is viewed. The peak object does not have a position defined for non-Q dimension.
```
I'm not certain about these suggested changes, but the singulars read weird without them:
```suggestion
- The peak overlay is now disabled when a non-Q axis is viewed. The peak object does not have a position defined for a non-Q dimension.
```

cailafinn reviewed (changes requested):
Just a couple of very minor grammar and formatting notes

sf1919 committed:
Revisions to Workbench notes following review

- Added missing words and fullstop
- Added missing formatting
- Added link to Superplot documentation

cailafinn reviewed (approved):
Issues addressed and looking good to me now :)
